node-src/tasks/prepare.ts

```diff
@@ -0,0 +1,206 @@
+import * as turbosnap from '@cli/turbosnap';
+import { readdirSync, readFileSync, statSync } from 'fs';
+import path from 'path';
+import semver from 'semver';
+import slash from 'slash';
+
+import { getFileHashes } from '../lib/getFileHashes';
+import { createTask, transitionTo } from '../lib/tasks';
+import { rewriteErrorMessage } from '../lib/utils';
+import { Context, Task } from '../types';
+import missingStatsFile from '../ui/messages/errors/missingStatsFile';
+import deviatingOutputDirectory from '../ui/messages/warnings/deviatingOutputDirectory';
+import {
+  bailed,
+  hashing,
+  initial,
+  invalid,
+  success,
+  traced,
+  tracing,
+  validating,
+} from '../ui/tasks/prepare';
+
+interface PathSpec {
```

Comment:
I'd recommend adding a comment here indicating why a simple `Path` isn't useful. Further, do you think it would be useful to work with a member that is a true `Path`, rather than a `string` (my assumption is you chose this for some reason, so I suspect your choice is valid, I'm just curious why).

Also, thoughts on the name? I don't particularly like `PathSpec`, because it's not very descriptive, but I can't come up with another one, other than something like `PathWithContent` or `ContentfulPath`.